common/components/common/upload/FileUploadButton.jsx

```diff
@@ -29,12 +30,29 @@ class FileUploadButton extends React.PureComponent<Props, State> {
   }
   
   render(): React$Node {
-    return (
-      <div>
-        <input type="button" value={this.props.buttonText} onClick={this._handleClick.bind(this)} />
-        <input ref="fileInput" type="file" style={{display:"none"}} accept={this.props.acceptedFileTypes} onChange={this._handleFileSelection.bind(this)} />
-      </div>
-    );
+    if(this.props.thumbnail_image == "true"){
```

Comment:
I think a better way to structure this component would be to have a buttonText prop for where we want a regular button, and a glyphIcon prop (taking in 'fa-plus' in this case) for where we want an icon.

A couple other points:
-Be sure to add new props to the Props type up above
-It's best to pass around the boolean constant true, not the string "true".  And use the === comparison operator on.